src/all.rs

```diff
@@ -57,6 +125,58 @@ impl<'a, Trait: ?Sized + TraitQuery> Iterator for ReadTableTraitsIter<'a, Trait>
     }
 }
 
+impl<'a, Trait: ?Sized + TraitQuery> Iterator for AddedReadTableTraitsIter<'a, Trait> {
+    type Item = &'a Trait;
+    fn next(&mut self) -> Option<Self::Item> {
+        // Iterate the remaining table components that are registered,
+        // until we find one that exists in the table.
+        let (column, meta) = unsafe { zip_exact(&mut self.components, &mut self.meta) }
+            .find_map(|(&component, meta)| self.table.get_column(component).zip(Some(meta)))?;
+
+        // SAFETY: we know that the table row is a valid index???
+        let column_ticks = unsafe { column.get_ticks_unchecked(self.table_row).deref() };
+        column_ticks
+            .is_added(self.last_change_tick, self.change_tick)
+            .then(|| unsafe {
+                // SAFETY ISSUE! SAFETY ISSUE! SAFETY ISSUE! Unlike in the write case, we do not have
+                // exclusive access! We have shared access to the entire column and ticks? This might be
+                // okay though because there cannot be any other accesses with write access at the same
+                // time?
```

Comment:
This is correct. `AddedReadTraits` registers read-access, so the scheduler prevents write-access from occurring. Same deal in the other places you have this comment.